vaultkeeper/vaultkeeper.py

```diff
@@ -129,9 +129,14 @@ def run(self):
                                shell=False
                                )
         while app.poll() is None:
-            self.renew_token()
-            self.renew_all()
-            time.sleep(self.configs.refresh_interval)
+            try:
+                app.wait(timeout=self.configs.refresh_interval)
+            except TimeoutExpired as tex:
+                self.logger.info('Renewing leases...')
+                self.renew_token()
+                self.renew_all()
+            else:
+                system.exit(0)
```

Comment:
I'm not entirely comfortable with `sys.exit()` in a method like this. Maybe we should `return app.returncode` and then pass that to `sys.exit()` in `main()`?

Reply:
Makes sense - I'll do that rather.